### ISSUE TITLE
`bun test --watch` does not detect new test files.

### ISSUE THREAD

**Issue by DaleSeo** [oven-sh/bun#8342]
### What version of Bun is running?

1.0.25

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

1. run `bun test --watch`
2. add a new file with some test code

### What is the expected behavior?

The new test file is detected and test cases in the file get exectued.

### What do you see instead?

The new tests are not inclluded. The terminal doesn't flick.

**Comment by Jarred-Sumner** [comment]
yeah, this is an issue with how the watcher currently works. We probably need special code for bun test to handle this.
